view.php

```diff
@@ -157,7 +157,13 @@
     if ($downloadown) {
         // Create new customcert issue record if one does not already exist.
         if (!$DB->record_exists('customcert_issues', array('userid' => $USER->id, 'customcertid' => $customcert->id))) {
-            \mod_customcert\certificate::issue_certificate($customcert->id, $USER->id);
+            /**
+             * Do not write a line in the table if the user is admin, teacher or manager and only looking at a student's certificate.
+             * Only students viewing their certificate are written as a row in the table, otherwise do nothing.
+             */
+            if(!has_capability('enrol/manual:enrol', $context)){
```

Comment:
I swear I commented on this already but can't see where it has gone. Just to repeat myself we can't use the capability `enrol/manual:enrol` as people can be enrolled in a course via numerous ways.